### PR TITLE
add flycheck-clojure

### PR DESCRIPTION
**Pull request by JAremko**
Finally made it work properly(maybe?, hopefully..) with Spacemacs.

### CODE REVIEW

**Comment by sdwolfz** [comment]
@WarFox Do you think you can take a look at this and tell us how it works for you?

**Comment by WarFox** [comment]
sure @sdwolfz, I'd be happy to test this. Thank you for making this @JAremko

**Comment by WarFox** [comment]
Tried this in my setup, but it doesn't seem to work

``` clojure
(defn fly-tests []
  (inc "foo")
  (map inc [1 2 3])
  (+ 3))
```

When I do `SPC e b` in a clojure-mode buffer with above code

I get

```
error in process filter: Wrong number of arguments: (4 . 4), 0 [2 times]
Error from syntax checker clojure-cider-eastwood: Done with no errors
```

No squiggly lines. Is there any configuration that I need to do?

**Comment by WarFox** [comment]
Thanks.
I just tried it, no changes. Same errors.

Should I add/change anything in profiles.clj?

This is what I have in my `~/.lein/profiles.clj`

```
{:user {:plugins [
                  ;; [cider/cider-nrepl "0.18.0"]
                  ;; [com.billpiel/sayid "0.0.16"]
                  [com.jakemccrary/lein-test-refresh "0.16.0"]
                  [jonase/eastwood "0.3.4"]
                  [lein-auto "0.1.3"]
                  [lein-cloverage "1.0.13"]
                  [lein-kibit "0.1.6"]
                  [lein-nvd "0.5.5"]
                  [lein-pprint "1.1.1"]
                  [lein-eftest "0.5.3"]
                  [nrepl/lein-nrepl "0.3.1"]
                  ;; [refactor-nrepl "2.4.0"]
                  ;; [venantius/ultra "0.5.2"]
                  ]
        :dependencies [
                       [alembic "0.3.2"]
                       [slamhound "1.3.1"]
                       ]
        :test-refresh {:notify-command ["terminal-notifier" "-title" "Tests" "-message"]
                       :quiet true
                       :changes-only true}
        }
 }
```

**Comment by JAremko** [comment]
@WarFox No. https://github.com/clojure-emacs/squiggly-clojure#dependencies-in-clojure 

But you can try with their simple project to be sure that it isn't some kind of deps conflict. https://github.com/clojure-emacs/squiggly-clojure/tree/master/sample-project

**Comment by JAremko** [comment]
@WarFox  thx for testing :heart:

**Comment by WarFox** [comment]
I tried it in the squiggly-clojure sample project, and it seems to work! 👏 

image

image

Good job 💯

**Comment by JAremko** [comment]
@sdwolfz  I think we can merge it. I added necessary warnings to the README.org and made linting disabled by default. :thinking:

File: layers/+lang/clojure/README.org [hunk 1]
```diff
@@ -184,6 +186,17 @@ Type ~SPC m h c~ to display the cheatsheet then type in some terms (space
 separated) to narrow down the list. For example, try typing in sort map to see
 some functions that deal with sorting maps.
 
+** Linting
+Linting works via flycheck and disabled by default.
+You can enablde it like this:
+
+#+BEGIN_SRC emacs-lisp
+  (setq clojure-auto-enable-linters t)
```
Comment (sdwolfz) [inline layers/+lang/clojure/README.org]: I think you meant `clojure-enable-linters` here.

**Comment by JAremko** [comment]
@sdwolfz  how about now :thinking:

**Pull request merged**